### PR TITLE
New Decisions Feature

### PR DESCRIPTION
**Pull request by dhaselhan**
* Allow multiple decisions per application, with new UI
* Add file upload per decision

### CODE REVIEW

File: alcs-api/src/application/application-decision/application-decision-outcome.entity.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { Entity } from 'typeorm';
+import { BaseCodeEntity } from '../../common/entities/base.code.entity';
+
+@Entity()
+export class ApplicationDecisionOutcome extends BaseCodeEntity {}
```
Comment (mhuseinov) [inline alcs-api/src/application/application-decision/application-decision-outcome.entity.ts:5]: ApplicationDecisionOutcomeType ?

File: alcs-api/src/application/application-decision/application-decision.controller.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    );
+    const mappedCodes = this.mapper.mapArray(
+      codes,
+      ApplicationDecisionOutcome,
+      ApplicationDecisionOutcomeDto,
+    );
+
+    return {
+      decisions: mappedDecisions,
+      codes: mappedCodes,
```
Comment (mhuseinov) [inline alcs-api/src/application/application-decision/application-decision.controller.ts:67]: why not include code into CodeController?

Comment (dhaselhan) [reply alcs-api/src/application/application-decision/application-decision.controller.ts:67]: Its only needed on a single page, I figure this was better than loading it all the time even when its not needed.

File: alcs-frontend/src/app/features/application/decision/decision.component.html [hunk 3]
```diff
@@ -1,10 +1,57 @@
...
-<section>
-  <h4>Decision Date</h4>
-  <div>
-    <app-inline-datepicker [selectedValue]="decisionDate" (save)="setDecisionDate($event)"></app-inline-datepicker>
+<div class="split">
+  <h3>Decision</h3>
+  <button mat-flat-button color="primary" (click)="onCreate()">+ New Decision</button>
+</div>
+<section class="decision" *ngFor="let decision of decisions; let i = index">
+  <div *ngIf="decisions.length - 1 !== i" class="recon-label">Recon</div>
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/application/decision/decision.component.html:6]: this recon label that is not recon 😅

File: alcs-api/src/application/application-decision/application-decision-outcome.entity.ts (hunk 1 above)
Comment (dhaselhan) [reply alcs-api/src/application/application-decision/application-decision-outcome.entity.ts]: Done! Renamed and updated migrations.

File: alcs-api/src/application/application-decision/application-decision.service.ts [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    const existingDecisions = await this.getByAppFileNumber(
+      applicationDecision.application.fileNumber,
+    );
+    if (existingDecisions.length === 0) {
+      await this.applicationService.createOrUpdate({
+        decisionDate: null,
+        fileNumber: applicationDecision.application.fileNumber,
+      });
+    } else {
+      await this.applicationService.createOrUpdate({
```
[the remark this answers is not in the document]
Comment (dhaselhan) [reply alcs-api/src/application/application-decision/application-decision.service.ts:175]: Created follow up ticket to split createOrUpdate into create / update.

**Review by mhuseinov — APPROVED** [review]
👍

**Pull request merged**